parser/src/parser/grammar.lalrpop

```diff
@@ -118,6 +120,40 @@ PeriodicColumn: PeriodicColumn = {
         PeriodicColumn::new(name, values),
 }
 
+// RANDOM VALUES
+// ================================================================================================
+
+RandomValues: RandomValues = {
+    "random_values" ":" <rand_values: ListElems<RandValuesDecl>> =>? match rand_values.len() {
+        1 => Ok(rand_values[0].clone()),
+        _ => Err(ParseError::User{
+            error: Error::ParseError(
+                InvalidRandomValues("No more than one set of random values can be declared".to_string())
+            )
+        })
+    }
+}
+
+RandValuesDecl: RandomValues = {
+    <name: Identifier> ":" <size: Size> => RandomValues::new(name, size, vec![]),
+    <_name: Identifier> ":" "[" "]" =>? Err(
+        ParseError::User{
+            error: Error::ParseError(
+                InvalidRandomValues(
+                    "Random Values section cannot be empty".to_string()
+                )
+            )
+        }
+    ),
+    <name: Identifier> ":" "[" <rand_vec: CommaElems<RandElem>> "]" => RandomValues::new(name, rand_vec.iter().map(|v| v.size()).sum(), rand_vec)
```

Comment:
Can these two be combined to return an error if `rand_vec` is empty or does that not work?

Reply:
Unfortunately, I couldn't get it to work like this